sim/mage/arcane_explosion.go

```diff
@@ -0,0 +1,38 @@
+package mage
+
+import (
+	"github.com/wowsims/cata/sim/core"
+)
+
+func (mage *Mage) registerArcaneExplosionSpell() {
+
+	mage.ArcaneExplosion = mage.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 1449},
+		SpellSchool:    core.SpellSchoolArcane,
+		ProcMask:       core.ProcMaskSpellDamage,
+		Flags:          SpellFlagMage | core.SpellFlagAPL,
+		ClassSpellMask: MageSpellArcaneExplosion,
+
+		ManaCost: core.ManaCostOptions{
+			BaseCost: 0.22,
+		},
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+		},
+
+		DamageMultiplier: 1,
+		CritMultiplier:   mage.DefaultSpellCritMultiplier(),
+		BonusCoefficient: 0.186,
+		ThreatMultiplier: 1 - 0.4*float64(mage.Talents.ImprovedArcaneExplosion),
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			for _, aoeTarget := range sim.Encounter.TargetUnits {
+				baseDamage := 0.368 * mage.ScalingBaseDamage
+				baseDamage *= sim.Encounter.AOECapMultiplier()
+				spell.CalcAndDealDamage(sim, aoeTarget, baseDamage, spell.OutcomeMagicHitAndCrit)
+			}
```

Comment:
Aoe spells like this are usually first all results calculated and then all results dealt in a 2nd loop. You can find many examples in other classes